src/runtime_src/core/include/experimental/xrt_module.h

```diff
@@ -85,14 +85,19 @@ class module : public detail::pimpl<module_impl>
    *   Parent module with instruction buffer to move into hwctx
    * @param hwctx
    *   Hardware context to associate with module
+   * @param ctrl_code_idx
+   *   index of control code inside the parent module
    *
    * Copy content of existing module into an allocation associated
    * with the specified hardware context.
+   * If module has multiple control codes, index is used to identify
+   * the control code that needs to be run.
+   * By default control code at zeroth index is picked
    *
    * Throws if module is not compatible with hardware context
    */
   XRT_API_EXPORT
-  module(const xrt::module& parent, const xrt::hw_context& hwctx);
+  module(const xrt::module& parent, const xrt::hw_context& hwctx, uint32_t ctrl_code_idx = 0);
```

Comment:
This is a breaking change.   Existing code will no longer work because the function with no argument is no longer present.  Avoid default arguments, just create a new ctor.

Reply:
As I was using default arg existing application should work right Soren.
Yeah I will use new constructor instead of default arg.

Reply:
Yes understood Soren, changed the code to have 2 constructors